server/utils.go

```diff
@@ -397,6 +399,41 @@ func versionToString(vers int) string {
 	return str
 }
 
+// rewriteToken attempts to match the original token
+// against the email, telephone number or login patterns.
+// On success, prepends the token with the corresponding prefix.
+func rewriteToken(orig string) string {
+	if orig == "" || strings.HasPrefix(orig, "basic:") ||
+			strings.HasPrefix(orig, "email:") || strings.HasPrefix(orig, "tel:") {
+		return orig
+	}
+	// Is it email?
+	if emailRegexp.MatchString(orig) {
+		return "email:" + orig
+	}
+	// TODO: pass region as a param.
+	if num, err := libphonenumber.Parse(orig, "US"); err == nil {
+		// It's a phone number.
+		return "tel:" + libphonenumber.Format(num, libphonenumber.E164)
+	}
+	// Does it look like a username/login?
+	runes := []rune(orig)
+	if len(runes) >= 3 && unicode.IsLetter(runes[0]) {
```

Comment:
It's fine for now but it would be much better to test against configured authenticators. Maybe add a method to authenticator to check the token and prepend with tag if it's valid. 

If you agree please add a TODO to that extent.

Reply:
Agreed. Added a todo.